pages/admin/users.tsx

```diff
@@ -0,0 +1,112 @@
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { requireAuthentication } from "@lib/auth";
+import { User } from "next-auth";
+import { logMessage } from "@lib/logger";
+import { getUsers } from "@lib/users";
+import { Button } from "@components/forms";
+import { useRefresh } from "@lib/hooks/useRefresh";
+import React from "react";
+import axios from "axios";
+import { TFunction, useTranslation } from "next-i18next";
+
+// We're filtering out all the undefined fields in getServerSideProps
+// Extending interface locally to be easier to read
+interface AuthenticatedUser extends User {
+  id: number;
+  admin: boolean;
+}
+
+const updateAdminValue = async (userID: number, isAdmin: boolean) => {
+  return await axios({
+    url: `/api/users`,
+    method: "POST",
+    data: {
+      userID,
+      isAdmin,
+    },
+    timeout: process.env.NODE_ENV === "production" ? 60000 : 0,
+  }).catch((e) => {
+    logMessage.error(e);
+  });
+};
+
+const AdminEnable = ({
+  isAdmin,
+  t,
+  user,
+}: {
+  isAdmin: boolean;
+  t: TFunction;
+  user: AuthenticatedUser;
+}) => {
+  const { refreshData, isRefreshing } = useRefresh([isAdmin]);
+  return (
+    <Button
```

Comment:
I think this is missing the `ariaProps` that you've added to the `<Button>` component above.

Reply:
In this case the Aria-props isn't necessary.  I've simply added the feature for future use.
Instead of using aria-props and creating a new element, in order to reference it's id, I've used a span within the button that is only screen-reader visible.